spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/test/java/io/awspring/cloud/s3/S3TemplateIntegrationTests.java

```diff
@@ -147,6 +149,25 @@ void deletesObjectByS3Url() {
 				.isThrownBy(() -> client.headObject(r -> r.bucket(BUCKET_NAME).key("key.txt")));
 	}
 
+	@Test
+	void listObjects() throws IOException {
+		client.putObject(r -> r.bucket(BUCKET_NAME).key("hello-en.txt"), RequestBody.fromString("hello"));
+		client.putObject(r -> r.bucket(BUCKET_NAME).key("hello-fr.txt"), RequestBody.fromString("bonjour"));
+		client.putObject(r -> r.bucket(BUCKET_NAME).key("bye.txt"), RequestBody.fromString("bye"));
+		
+		List<S3Resource> resources = s3Template.listObjects(BUCKET_NAME, "hello");
+		assertThat(resources.size()).isEqualTo(2);
+		
+		// According to the S3Client doc : "Objects are returned sorted in an ascending order of the respective key names in the list."
+		try (InputStream is = resources.get(0).getInputStream(); InputStream is2 = resources.get(1).getInputStream()) {
+			String result = StreamUtils.copyToString(is, StandardCharsets.UTF_8);
```

Comment:
It is just suggetion, not necessary to addressed. For assertion we can use asserj extracting and map functions to extract and convert `InputStream` to string and then compare the string and that to if they are in order.

Reply:
Hello @kunalvarpe !
Thanks for your review!
Like that ?
```
		assertThat(resources).extracting(S3Resource::getInputStream).map(is -> new String(is.readAllBytes(), StandardCharsets.UTF_8))
			.containsExactly("hello", "bonjour");